app/viewmodels/helpers/ItemQuantityHelper.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.helpers
+
+import models.GoodsType
+import models.response.referenceData.CnCodeInformation
+import play.api.i18n.Messages
+import utils.ExciseProductCodeHelper
+
+class ItemQuantityHelper {
+
+  def heading(goodsType: GoodsType, cnCodeInfo: CnCodeInformation)(implicit messages: Messages): String = {
+    val key: String = s"itemQuantity.heading.${cnCodeInfo.exciseProductCode}"
+    if (messages.isDefinedAt(key)) {
+      messages(key)
+    } else {
+      messages("itemQuantity.heading", goodsType.toSingularOutput())
+    }
+  }
+
+  def paragraph(cnCodeInfo: CnCodeInformation)(implicit messages: Messages): Option[String] = {
+    val key: String = s"itemQuantity.paragraph.${cnCodeInfo.exciseProductCode}"
+    Option.when(messages.isDefinedAt(key))(messages(key))
+  }
+
+  def label(cnCodeInfo: CnCodeInformation)(implicit messages: Messages): String = {
+    val key: String = s"itemQuantity.label.${cnCodeInfo.exciseProductCode}"
+    if (messages.isDefinedAt(key)) {
+      messages(key)
+    } else {
+      ""
```

Comment:
Is it right that we wouldn't output anything for the label if the message key doesn't exist? Would this ever happen?

Reply:
I don't think we can do it like this and output an empty label. Because that will cause an accessibility issue. 

The label would have to have the same question as the heading/title and the `visuallyhidden` class would need to be added to it.

Reply:
Might be better to return an `Option` here, and then handle that option in the view or a viewhelper to decide how to render the label.

Reply:
Refactored